feel/feeldiscr/functionspacebase.hpp

```diff
@@ -92,32 +99,46 @@ class FEELPP_EXPORT FunctionSpaceBase : public CommObject
      */
     //@{
 
+    // Get unique instance name.
+    const std::string& instanceName() const
+    {
+        return M_instance_name;
+    }
+
+    // Get unique instance number.
+    const uint16_type& instanceNumber() const
+    {
+        return M_instance_number;
+    }
 
     //@}
 
     /** @name  Mutators
      */
     //@{
 
-
     //@}
 
-    /** @name  Methods
-     */
-    //@{
+    class ElementBase {};
 
+private:
+    //! Private Methods
+    //! @{
 
-    //@}
+    // Simulation info observer notifications.
+    virtual const pt::ptree journalNotify() const override = 0;
 
-    class ElementBase {};
+    //! @}
 
 protected:
 
-private:
- 
-};
+    // Unique instance name.
+    std::string M_instance_name;
+    uint16_type M_instance_number;
 
```

Comment:
I think is too intrusive and it can be duplicated be a lot of.
We can define a class ObserverObject (like commObject) and  FunctionSpaceBase inherits of ObserverObject